hail/python/hailtop/batch/batch.py

```diff
@@ -149,6 +151,64 @@ def __init__(self,
 
         self._cancel_after_n_failures = cancel_after_n_failures
 
+        self._python_function_defs: Dict[int, Callable] = {}
+        self._python_function_files: Dict[int, Optional[_resource.InputResourceFile]] = {}
+
+    def _get_python_function(self, function_id: int) -> Callable:
+        function = self._python_function_defs.get(function_id, None)
+        if function is None:
+            raise ValueError(
+                f"No Python function registered with id: {function_id}."
+            )
+        return function
+
+    def _get_python_function_file(self, function_id: int) -> _resource.InputResourceFile:
+        try:
+            file = self._python_function_files[function_id]
+            if file is None:
+                raise ValueError(
+                    f"No file found for Python function registered with id: {function_id}."
+                )
+            return file
+        except KeyError as error:
+            raise ValueError(
+                f"No Python function registered with id: {function_id}."
+            ) from error
+
+    def _register_python_function(self, function: Callable) -> int:
+        function_id = id(function)
+        try:
+            self._get_python_function(function_id)
+        except ValueError:
+            self._python_function_defs[function_id] = function
+        return function_id
```

Comment:
Let's remove the try-except and just run the line that sets the value in the dictionary, shouldn't matter if it's already in there

Reply:
Removed!